api/spec/requests/spree/api/orders_controller_spec.rb

```diff
@@ -815,6 +815,38 @@ module Spree
           expect(response.status).to eq 201
           expect(json_response["user_id"]).to eq(user.id)
         end
+
+        context 'with payment', js: true do
+          let(:params) do
+            {
+              payments: [{
+                amount: '10.0',
+                payment_method: payment_method.name,
+                source: {
+                  month: "01",
+                  year: Date.today.year.to_s.last(2),
+                  cc_type: "123",
+                  last_digits: "1111",
+                  name: "Credit Card"
+                }
+              }]
+            }
+          end
+          let!(:payment_method) { create(:payment_method) }
+
+          context "with source" do
+            it "creates a payment" do
+              post spree.api_orders_path, params: { order: params }
+              expect(response.status).to eq 201
```

Comment:
I think it's worth expanding a bit the expectations for this spec. The title says that `it creates a payment` but actually here we just check that the response is successful. I think adding something like this may be an improvement: 
```
payment = json_response['payments'].first
expect(payment['amount']).to eql "10.0"
expect(payment['source']['last_digits']).to eql "1111" 
```

Reply:
done here https://github.com/solidusio/solidus/pull/3066/files#diff-72468e52a59aedfe8a3443597f1a6f3aR836